src/software/power/charger.cpp

```diff
@@ -0,0 +1,50 @@
+#include "charger.h"
+
+volatile bool Charger::flyback_fault    = false;
+void (*Charger::charge_done_callback)() = NULL;
+
+Charger::Charger()
+{
+    pinMode(HV_SENSE, INPUT);
+    pinMode(FLYBACK_FAULT, INPUT);
+    pinMode(CHRG_DONE, INPUT);
+    pinMode(CHRG, OUTPUT);
+
+    attachInterrupt(digitalPinToInterrupt(CHRG_DONE), chargeDone, RISING);
```

Comment:
see: https://discord.com/channels/711281059857432668/764565540819042342/980927731527208970
done pin goes low when its done charging, this should be FALLING